test/assignment.coffee

```diff
@@ -227,7 +227,35 @@ test "destructuring assignment with objects and splats", ->
   {a: b: [y, z...]} = obj
   eq a, y
   arrayEq [b,c,d], z
-
+  
+test "destructuring assignment with objects and splats: ES2015", ->
+  obj = {a:1, b:2, c:3, d:4, e:5}
+  {a:x, b, r...} = obj
+  {a:y, c..., d} = obj
```

Comment:
The rest element has to be last, according to Babel: http://babeljs.io/repl/#?babili=false&evaluate=false&lineWrap=false&presets=es2015%2Cstage-2%2Cstage-3&targets=&browsers=&builtIns=false&code=const%20%7Ba%3Ay%2C%20...c%2C%20d%7D%20%3D%20obj&experimental=false&loose=false&spec=false&playground=true

I guess CoffeeScript should support it anywhere, though, just like for arrays? I guess we just need to move the splat last in the compiled JS.